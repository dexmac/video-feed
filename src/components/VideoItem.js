import React, { Component } from "react";
import PropTypes from "prop-types";
import ReactPlayer from "react-player";
import VideoItemHeader from "./VideoItemHeader";
import { ERR_LOADING_VIDEO_PREFIX } from "../constants";

class VideoItem extends Component {
  state = {
    shouldShowVideo: true,
    errorString: ""
  };

  onVideoError = url => {
    this.setState({ shouldShowVideo: false, errorString: url });
  };

  render() {
    const item = this.props.item;

    let VideoPlayer;
    if (this.state.shouldShowVideo) {
      VideoPlayer = (
        <ReactPlayer
          className="react-player"
          url={item ? item.url : null}
          key={item ? item.videoId : null}
          width="100%"
          height="100%"
          onError={() => this.onVideoError(item ? item.url : null)}
          playing
          muted
          playsinline
          config={{ file: { forceVideo: true } }}
        />
      );
    } else {
      // On Video Load Error
      VideoPlayer = (
        <div className="video-load-error-text">
          {ERR_LOADING_VIDEO_PREFIX} : {this.state.errorString}
        </div>
      );
    }

    return (
      <div className="video-item">
        <VideoItemHeader
          title={item && item.title ? item.title : ""}
          views={item && item.views ? item.views : null}
        />
        <div className="player-wrapper">{VideoPlayer}</div>
      </div>
    );
  }
}

VideoItem.propTypes = {
  item: PropTypes.shape({
    url: PropTypes.string,
    source: PropTypes.string,
    videoId: PropTypes.string,
    title: PropTypes.string,
    views: PropTypes.number
  })
};
export default VideoItem;
