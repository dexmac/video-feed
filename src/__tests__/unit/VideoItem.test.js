import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import VideoItem from "../../components/VideoItem";

it("renders correctly", () => {
    const tree = shallow(<VideoItem />);
    expect(tree).toMatchSnapshot();
});
