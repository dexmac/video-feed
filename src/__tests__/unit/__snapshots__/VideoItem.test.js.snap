// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="video-item"
>
  <VideoItemHeader />
  <div
    className="player-wrapper"
  >
    <ReactPlayer
      className="react-player"
      config={
        Object {
          "dailymotion": Object {
            "params": Object {
              "api": 1,
              "endscreen-enable": false,
            },
            "preload": false,
          },
          "facebook": Object {
            "appId": "1309697205772819",
          },
          "file": Object {
            "attributes": Object {},
            "forceAudio": false,
            "forceDASH": false,
            "forceHLS": false,
            "hlsOptions": Object {},
            "tracks": Array [],
          },
          "mixcloud": Object {
            "options": Object {
              "hide_cover": 1,
            },
          },
          "soundcloud": Object {
            "options": Object {
              "buying": false,
              "download": false,
              "liking": false,
              "sharing": false,
              "show_comments": false,
              "show_playcount": false,
              "visual": true,
            },
          },
          "twitch": Object {
            "options": Object {},
          },
          "vimeo": Object {
            "playerOptions": Object {
              "autopause": false,
              "autoplay": false,
              "byline": false,
              "portrait": false,
              "title": false,
            },
            "preload": false,
          },
          "wistia": Object {
            "options": Object {},
          },
          "youtube": Object {
            "playerVars": Object {
              "autoplay": 0,
              "iv_load_policy": 3,
              "modestbranding": 1,
              "playsinline": 1,
              "rel": 0,
              "showinfo": 0,
            },
            "preload": false,
          },
        }
      }
      controls={false}
      height="100%"
      key="null"
      loop={false}
      muted={true}
      onBuffer={[Function]}
      onDuration={[Function]}
      onEnded={[Function]}
      onError={[Function]}
      onPause={[Function]}
      onPlay={[Function]}
      onProgress={[Function]}
      onReady={[Function]}
      onSeek={[Function]}
      onStart={[Function]}
      playbackRate={1}
      playing={true}
      playsinline={true}
      progressInterval={1000}
      style={Object {}}
      url={null}
      volume={null}
      width="100%"
      wrapper="div"
    />
  </div>
</div>
`;
